src/biome/text/api_new/pipeline.py

```diff
@@ -632,103 +693,90 @@ async def status():
+                    for automatically generate data iterator sorting keys
+                    """
+                    required, _ = split_signature_params_by_predicate(
+                        pipeline.head.forward,
+                        lambda p: p.default == inspect.Parameter.empty,
+                    )
+                    return [p.name for p in required] or [None]
+
+                iterator_config = {
+                    "batch_size": trainer.batch_size,
+                    "max_instances_in_memory": max(
+                        trainer.batch_size * trainer.in_memory_batches,
+                        trainer.batch_size,
+                    ),
+                    "cache_instances": trainer.cache_instances,
+                    "type": "basic",
+                }
 
-        # Force clean folder for run fine tuning properly
-        # TODO: reuse vocab
-        shutil.rmtree(config.output, ignore_errors=True)
-        # _recover_output_folder(config.output, fine_tune_params)
+                if trainer.data_bucketing:
+                    iterator_config.update(
+                        {
+                            "sorting_keys": [
+                                [
+                                    _forward_inputs()[0],
+                                    "list_num_tokens"
+                                    if isinstance(
+                                        pipeline.model.encoder, TimeDistributedEncoder
+                                    )
+                                    else "num_tokens",
+                                ]
+                            ],
+                            "type": "bucket",
+                        }
+                    )
+
+                return iterator_config
+
+            base_config = {
+                "config": pipeline.config.as_dict(),
+                "type": __default_impl__.__name__,
+            }
+            allennlp_config = {
+                "trainer": trainer_configuration(config.trainer),
+                "iterator": iterator_configuration(pipeline, config.trainer),
+                "dataset_reader": base_config,
+                "model": base_config,
+                "train_data_path": config.training,
+                "validation_data_path": config.validation,
+                "test_data_path": config.test,
+            }
+            return copy.deepcopy({k: v for k, v in allennlp_config.items() if v})
 
-        fine_tune_model(
-            model=pipeline._model,
-            params=fine_tune_params,
+        pipeline._model.launch_experiment(
+            params=Params(allennlp_configuration(pipeline, config)),
             serialization_dir=config.output,
-            extend_vocab=True,  # TODO: Allow parameterize
+            extend_vocab=config.extend_vocab,
             file_friendly_logging=True,
-        )
+        )  # pylint: disable=protected-access,
 
         return pipeline.__class__(
```

Comment:
A bit unrelated, but isn't the config ignored (taken from archive) when constructing the `Pipeline` from pretrained? I had left a comment  in the __init__ method, regarding the signature which is a bit confusing with two Optionals, not sure if it's too relevant though.

Reply:
Yes, the arguments are exclusives. So we just need to include some argument validations in top of `__init__` for avoid confussions